test/test_esapi.py

```diff
@@ -1435,6 +1436,63 @@ def test_hmac(self):
         with self.assertRaises(TypeError):
             self.ectx.hmac(primaryHandle, inData, TPM2_ALG.SHA256, session3=45.6)
 
+    def test_mac(self):
```

Comment:
Oh it works because your raising an Exception of NotImplementedError?

Perhaps this should decorate this function as:
```python
    @pytest.mark.skipif(
        not _lib_version_atleast("tss2-esapi", "4.0.0"), reason="Requires tss2-esapi 4.0 or greater"
    )
```